src/ott/problems/nn/dataset.py

```diff
@@ -0,0 +1,155 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Neural OT datasets."""
+
+import dataclasses
+from typing import Any, Iterable, Iterator, Literal, NamedTuple, Tuple
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+__all__ = ['create', 'Dataloaders', 'GaussianMixture']
+
+Name_t = Literal['gaussian_simple', 'gaussian_circle', 'gaussian_square_five',
+                 'gaussian_square_four']
+
+
+class Dataloaders(NamedTuple):
```

Comment:
I think slightly cleaner would be to call this `Dataset` and have only 2 properties:
`{source,target}_iter` + `create` return 2 datasets (1 for train, other for validation).

Reply:
I wrote it like this so that `create` returns an object that can be directly passed into the solver:

```
dataloaders = dataset.create('gaussian_simple', 'gaussian_circle')
learned_potentials = neural_dual_solver(*dataloaders)
```

But I agree it does seem slightly cleaner to group the source/target samplers together. Do you have a preference for how I modify it? We could also update the interface of the `neural_dual` to take a similar grouping

Reply:
```
train_dataloaders, val_dataloaders = dataset.create('gaussian_simple', 'gaussian_circle')
learned_potentials = neural_dual_solver(*train_dataloaders, *val_dataloaders)
```